packages/@aws-cdk/toolkit-lib/lib/api/drift/drift.ts

```diff
@@ -0,0 +1,87 @@
+import { format } from 'util';
+import type { DescribeStackDriftDetectionStatusCommandOutput, DescribeStackResourceDriftsCommandOutput } from '@aws-sdk/client-cloudformation';
+import { ToolkitError } from '../../toolkit/toolkit-error';
+import type { ICloudFormationClient } from '../aws-auth/private';
+import type { IoHelper } from '../io/private';
+import { IO } from '../io/private';
+
+/**
+ * Detect drift for a CloudFormation stack and wait for the detection to complete
+ *
+ * @param cfn - a CloudFormation client
+ * @param ioHelper - helper for IO operations
+ * @param stackName - the name of the stack to check for drift
+ * @returns the CloudFormation description of the drift detection results
+ */
+export async function detectStackDrift(
+  cfn: ICloudFormationClient,
+  ioHelper: IoHelper,
+  stackName: string,
+): Promise<DescribeStackResourceDriftsCommandOutput> {
+  await ioHelper.notify(IO.CDK_TOOLKIT_I4506.msg(format('Starting drift detection for stack %s...', stackName)));
```

Comment:
Please either include a useful payload here (maybe `StackSelectionDetails`?) or make it a default message.